CHANGELOG.md

```diff
@@ -58,6 +58,7 @@ And please only add new entries to the top of this list, right below the `# Unre
 - **Breaking:** On X11, device events are now ignored for unfocused windows by default, use `EventLoopWindowTarget::set_device_event_filter` to set the filter level.
 - Implemented `Default` on `EventLoop<()>`.
 - Implemented `Eq` for `Fullscreen`, `Theme`, and `UserAttentionType`.
+- All platforms can now be relied on to emit a `Resumed` event. Applications are recommended to lazily initialize graphics state and windows on first resume for portability.
```

Comment:
I've never dynamically created windows inside the `Resumed` event yet (nor anywhere else), is that a thing?

Reply:
Yeah - for sure that's at least is fine with the Android, Windows and X11 backends.

The Egui app I linked above does that, as do a bunch of smaller test apps I have. This Android enabling branch I have for Bevy also does the same: https://github.com/rib/bevy/tree/android-enabling (deferring all render state + window creation until after resumed).

Although the Android backend can technically let you create a (hollow) Window even before we have a surface view that conceptually really doesn't sit well with me so I'd much rather wait until it's possible to create a render surface too before making the winit window.

Thinking out loud for ways of making Winit usage portable by default then another idea we could consider here is that create_window() can be explicitly documented to fail before the application reaches it's first `Resumed` event (and would fail consistently across platforms).

Reply:
Yeah, either that or do the `WindowEvent::Created` dance which hands over control of newly created windows from `winit` to the user, and disallow windows to be acquired elsewhere?

Not sure what such an API would look like though, but yes it's technically correct that an Android _window_ is only created after `Resumed` (and I think the functions like `fn size()` on `window` fail one way or another before that) and destroyed again after `Suspend`.

As mentioned above, not sure what `WindowEvent::Destroyed` means in the case of Android, and if we should use that instead of `::Suspended`.

Reply:
> Yeah, either that or do the `WindowEvent::Created` dance which hands over control of newly created windows from `winit` to the user, and disallow windows to be acquired elsewhere?
> 
> Not sure what such an API would look like though, but yes it's technically correct that an Android _window_ is only created after `Resumed` (and I think the functions like `fn size()` on `window` fail one way or another before that) and destroyed again after `Suspend`.
> 
> As mentioned above, not sure what `WindowEvent::Destroyed` means in the case of Android, and if we should use that instead of `::Suspended`.

I think I've addressed how I currently see this in this comment: https://github.com/rust-windowing/winit/pull/2331#discussion_r895666043

Going in this kind of direction I'd probably call your `WindowEvent::Created` `WindowEvent::SurfaceCreated` and the opposite `WindowEvent::SurfaceDestroyed` where it would become much easier to disambiguate from `WindowEvent::Destroyed`.

The challenge with that direction though is that Winit isn't currently responsible of render surfaces - it assumes all OSs can manually allocate surfaces for a window which basically isn't true on Android. That would make it hard know how these events would be in any way cross-platform. On Android it would make sense that these would deliver a handle for the the SurfaceView but there's no equivalent on other platforms - on other platforms it's just left to APIs like wgpu to allocate surfaces and it's assumed that they are fully managed/owned outside Winit.

Technically it's quite tricky, I'd say, to have an abstraction like Winit that doesn't claim any responsibility for render surfaces and for synchronizing swap buffers too so I'd certainly think it's worth questioning if that's the best design, but yeah at this point we're opening a much bigger can of worms, so it probably makes sense to discuss this stuff in a separate issue.

Reply:
> On Android it would make sense that these would deliver a handle for the the SurfaceView but there's no equivalent on other platforms

`RawWindowHandle`?